blocks/cards/cards.js

```diff
@@ -6,12 +6,36 @@ export default function decorate(block) {
   [...block.children].forEach((row) => {
     const li = document.createElement('li');
     while (row.firstElementChild) li.append(row.firstElementChild);
+
+    // Assign classes to div elements before wrapping in a link
     [...li.children].forEach((div) => {
-      if (div.children.length === 1 && div.querySelector('picture')) div.className = 'cards-card-image';
-      else div.className = 'cards-card-body';
+      if (div.children.length === 1 && div.querySelector('picture')) {
+        div.className = 'cards-card-image';
+      } else {
+        div.className = 'cards-card-body';
+      }
     });
+
+    // Check if the third column exists and contains a link
+    const thirdColumn = li.children[2]; // Get the third child (0-based index)
+    if (thirdColumn && thirdColumn.querySelector('a')) {
+      const link = thirdColumn.querySelector('a'); // Get the link
+      const { href } = link; // Extract the href attribute using destructuring
+
+      // Remove the text content of the link
+      link.textContent = ''; // Clears any text inside the <a> tag
+
+      // Wrap the entire <li> content in the link
+      const wrapperLink = document.createElement('a');
+      wrapperLink.href = href;
+      wrapperLink.style.display = 'block'; // Ensure the link covers the entire <li>
```

Comment:
I think we can simplify this better for authors.
1. Instead of needing a 3rd column, put your card link as the last item of the 2nd column.
2. linking the URL itself will [prevent the decorateButtons function from being applied (because the link text === link URL) ](https://github.com/aemsites/ams-lts/blob/main/scripts/aem.js#L397)
3. In your cards.js, you can check for if the last child of that 2nd card div is a link, and if (a.href !== a.textContent) {
-- your wrapperLink logic
}
4. As a general rule, try to refrain from adding CSS with JS. It's better to add a class name at line 31 for your new wrapperLink and put the display:block in cards.css. 

I have altered your draft page by adding a new set of cards with the 3 conditions an author might use.